synchronizer/synchronizer.go

```diff
@@ -463,21 +465,35 @@ func (d *Synchronizer) SyncCycle(customerDbUriFlag string) (bool, error) {
 		}
 
 		if crawler.SEER_CRAWLER_DEBUG {
-			log.Printf("Read batch key: %s", path)
+			log.Printf("Read batch key: %s", paths)
 		}
 		log.Println("Last block of current chank: ", lastBlockOfChank)
 
 		// Read the raw data from the storage for current path
-		rawData, readErr := d.StorageInstance.Read(path)
+		rawData, readErr := d.StorageInstance.ReadFilesAsync(paths, d.threads)
 		if readErr != nil {
 			return isEnd, fmt.Errorf("error reading raw data: %w", readErr)
 		}
 
 		log.Printf("Read %d users updates from the indexer db in range of blocks %d-%d\n", len(updates), d.startBlock, lastBlockOfChank)
 		var wg sync.WaitGroup
+		var errWg sync.WaitGroup
+		errWg.Add(1)
 
-		sem := make(chan struct{}, 5)  // Semaphore to control concurrency
-		errChan := make(chan error, 1) // Buffered channel for error handling
+		sem := make(chan struct{}, d.threads) // Semaphore to limit the number of concurrent goroutines
+		errChan := make(chan error)           // Channel to collect errors from goroutines
```

Comment:
I remember there was limited chan error `errChan := make(chan error, 1` otherwise it wont throw exceptions correctly